opencompass/openicl/icl_retriever/icl_zero_retriever.py

```diff
@@ -22,7 +22,7 @@ class ZeroRetriever(BaseRetriever):
     def __init__(self, dataset, ice_eos_token: Optional[str] = '') -> None:
         super().__init__(dataset, '', ice_eos_token, 0)
 
-    def retrieve(self, id_list: List[int]) -> List[List]:
+    def retrieve(self, id_list: List[int] = []) -> List[List]:
```

Comment:
Although `id_list` is not used in this func, it seems not recommend to use list as defaults

Reply:
https://github.com/open-compass/opencompass/issues/338

This PR is originated from here. I found it error-prone, thus use the default value to skip the redundant argument error.

Reply:
```suggestion
    def retrieve(self, id_list: Optional[List[int]] = None) -> List[List]:
```
Might use None as defaults?